### PR TITLE
Docs: Update Loki docs with new syntax and features

### LINKED ISSUE #19303: Docs: Update Loki docs
- Loki can be used for annotation
- Update logQl example and explanations.

### PR DESCRIPTION
**Pull request by aocenas**
Closes: https://github.com/grafana/grafana/issues/19303

### CODE REVIEW

File: docs/sources/features/datasources/loki.md [hunk 1]
```diff
@@ -111,11 +124,16 @@ log message you're interested in.
...
+
+Checkout the [Templating](/reference/templating) documentation for an introduction to the templating feature and the different types of template variables.
 
 ## Annotations
 
-Annotations are not yet supported by Loki.
+You can use any non metric Loki query as a source of annotations. Log content will be used annotation text so
+there is no need for additional mapping.
+
+> Note: Annotations for Loki are only available in Grafana v6.4+
```
Comment (aocenas) [inline docs/sources/features/datasources/loki.md]: Cargo culted the note here but are they valuable? We have version switcher in the docs site so I would assume this should be just current docs and not having them separately versioned as for older version these things just won't show.

File: docs/sources/features/datasources/loki.md [hunk 2]
```diff
@@ -111,11 +124,16 @@ log message you're interested in.
...
 
-Template variables are not yet supported by Loki.
+Instead of hard-coding things like server, application and sensor name in you metric queries you can use variables in their place. Variables are shown as dropdown select boxes at the top of the dashboard. These dropdowns makes it easy to change the data being displayed in your dashboard.
+
+Checkout the [Templating](/reference/templating) documentation for an introduction to the templating feature and the different types of template variables.
 
 ## Annotations
 
-Annotations are not yet supported by Loki.
+You can use any non metric Loki query as a source of annotations. Log content will be used annotation text so
```
Comment (aocenas) [inline docs/sources/features/datasources/loki.md]: Not sure if it makes sense to mention `non metric Loki query` here but seems like it's going to be important distinction.

File: docs/sources/features/datasources/loki.md [hunk 3]
```diff
@@ -88,13 +87,27 @@ After writing the Log Stream Selector, you can filter the results further by wri
...
+The following filter types are currently supported:
+
+* `|=` line contains string.
+* `!=` line does not contain string.
+* `|~` line matches regular expression.
+* `!~` line does not match regular expression.
+
+> Note: For more details about LogQL, Loki's query language, see it's [documentation](https://github.com/grafana/loki/blob/master/docs/querying.md)
 
 ## Live Tailing
```
Comment (marcusolsson) [inline docs/sources/features/datasources/loki.md]: Sentence case for headings:

```
## Live tailing
```

File: docs/sources/features/datasources/loki.md [hunk 4]
```diff
@@ -88,13 +87,27 @@ After writing the Log Stream Selector, you can filter the results further by wri
...
 
-To view your logs live as they are added, choose `Live` from the refresh dropdown, and you should see your logs be displayed in real time.
+Loki supports Live tailing which displays logs in real time. This feature is supported in [Explore](/features/explore/#loki-specific-features)
```
Comment (marcusolsson) [inline docs/sources/features/datasources/loki.md]: real-time is hyphenated

File: docs/sources/features/datasources/loki.md [hunk 5]
```diff
@@ -111,11 +124,16 @@ log message you're interested in.
 
 ## Templating
 
-Template variables are not yet supported by Loki.
+Instead of hard-coding things like server, application and sensor name in you metric queries you can use variables in their place. Variables are shown as dropdown select boxes at the top of the dashboard. These dropdowns makes it easy to change the data being displayed in your dashboard.
```
Comment (marcusolsson) [inline docs/sources/features/datasources/loki.md]: Missing `r`
> ... in your metric ...

Missing comma
> metric queries, you can use 

Drop-down box is hyphenated
> ... shown as drop-down boxes ...

Avoid using drop-down as a noun:
> These drop-down boxes makes it ...

File: docs/sources/features/datasources/loki.md [hunk 6]
```diff
@@ -111,11 +124,16 @@ log message you're interested in.
...
+
+Checkout the [Templating](/reference/templating) documentation for an introduction to the templating feature and the different types of template variables.
```
Comment (marcusolsson) [inline docs/sources/features/datasources/loki.md]: Simplify to:
> See the documentation on [Templating](/reference/templating) ...

Missing comma
> templating feature, and the different types

File: docs/sources/features/datasources/loki.md (hunk 2 above)
Comment (marcusolsson) [reply docs/sources/features/datasources/loki.md]: non-metric should be hyphenated

Comment (marcusolsson) [reply docs/sources/features/datasources/loki.md]: Correction:
> ... source for annotations ...

Comment (marcusolsson) [reply docs/sources/features/datasources/loki.md]: - Missing 'as'?
- Missing comma

> Log content will be used as annotation text, so there is no need for additional mapping.

**Review by marcusolsson — DISMISSED** [review]
There are some hard wraps in the file, that you might want to fix as well.

File: docs/sources/features/datasources/loki.md [hunk 7]
```diff
@@ -38,10 +38,7 @@ Just add it as a data source and you are ready to query your log data in [Explor
 
 ## Querying Logs
 
-Querying and displaying log data from Loki is available via [Explore](/features/explore).
-Select the Loki data source, and then enter a log query to display your logs.
-
-> Viewing Loki data in dashboard panels is not supported yet, but is being worked on.
+Querying and displaying log data from Loki is available via [Explore](/features/explore), and with [Table panel](/features/panels/table_panel/) in dashboards. Select the Loki data source, and then enter a log query to display your logs.
```
Comment (davkal) [inline docs/sources/features/datasources/loki.md]: Should be the logs panel now, no?

File: docs/sources/features/datasources/loki.md [hunk 8]
```diff
@@ -88,13 +85,28 @@ After writing the Log Stream Selector, you can filter the results further by wri
 
 Example queries:
 
-* `{job="mysql"} error`
-* `{name="kafka"} tsdb-ops.*io:2003`
-* `{instance=~"kafka-[23]",name="kafka"} kafka.server:type=ReplicaManager`
+* `{job="mysql"} |= error`
```
Comment (davkal) [inline docs/sources/features/datasources/loki.md]: All filter expressions need quotation marks, even single words, e.g., `|= "error"`

File: docs/sources/features/datasources/loki.md (hunk 1 above)
Comment (davkal) [reply docs/sources/features/datasources/loki.md]: I wonder if we should limit the display of annotation text somehow, so it's not going to blow up a tooltip.

**Review by davkal — CHANGES_REQUESTED** [review]
Looks great. Only blocker is the filter expression syntax.

File: docs/sources/features/datasources/loki.md (hunk 1 above)
Comment (aocenas) [reply docs/sources/features/datasources/loki.md]: > I wonder if we should limit the display of annotation text somehow, so it's not going to blow up a tooltip.

Well I agree with that, but wanted some feedback to the note with grafana version here and whether to have it in docs :D

File: docs/sources/features/datasources/loki.md (hunk 7 above)
Comment (aocenas) [reply docs/sources/features/datasources/loki.md]: Not sure is it still alpha? If so I am not sure if it should be mentioned here or not.

File: docs/sources/features/datasources/loki.md (hunk 1 above)
Comment (davkal) [reply docs/sources/features/datasources/loki.md]: +1 on cargo culting in this case :p

File: docs/sources/features/datasources/loki.md (hunk 7 above)
Comment (davkal) [reply docs/sources/features/datasources/loki.md]: Good point, let's change this part in the PR that brings this out of alpha.

**Review by davkal — APPROVED** [review]

**Comment by davkal** [comment]
Looks good from my side.

**Comment by aocenas** [comment]
@marcusolsson dismissed your review so I could merge, but all should have been fixed so hope thats fine :)

**Pull request merged**